pkg/lib/schemaloader/embedded.go

```diff
@@ -0,0 +1,100 @@
+package schemaloader
+
+import (
+	"embed"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/xeipuuv/gojsonschema"
+	"sigs.k8s.io/yaml"
+)
+
+// Schema is an interface that defines methods for validating data against a JSON schema.
+type Schema interface {
+	// ValidateFile validates the contents of a file specified by path against the schema.
+	ValidateFile(path string) (*gojsonschema.Result, error)
+
+	// ValidateReader validates the contents read from an io.Reader against the schema.
+	ValidateReader(r io.Reader) (*gojsonschema.Result, error)
+
+	// ValidateBytes validates the contents of a byte slice against the schema.
+	ValidateBytes(b []byte) (*gojsonschema.Result, error)
+}
+
+// NewEmbeddedSchema returns a Schema implementation that uses an embedded filesystem to contain a JSON schema.
+func NewEmbeddedSchema(fs embed.FS, root string) (Schema, error) {
+	schema, err := SchemaLoader(fs, root)
+	if err != nil {
+		return nil, err
+	}
+	return &embeddedSchema{
+		schema: schema,
+	}, nil
+}
+
+// embeddedSchema is an implementation of the Schema interface that uses a compiled JSON schema for validation.
+type embeddedSchema struct {
+	schema *gojsonschema.Schema
+}
+
+// ValidateFile validates the contents of a file specified by path against the schema.
+func (e *embeddedSchema) ValidateFile(path string) (*gojsonschema.Result, error) {
+	// Ensure we are given a json or yaml file only.
+	fileExt := filepath.Ext(path)
+	if !(fileExt == ".json" || fileExt == ".yaml" || fileExt == ".yml") {
+		return nil, fmt.Errorf("file extension (%s) not supported. The file must end in either .yaml, .yml or .json", fileExt)
```

Comment:
not that i care that much ... but why do we care? either it unmarshals or it doesn't

Reply:
I suppose we don't need to care, you're right. I can drop this check.